streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -448,29 +441,37 @@ private Topology setupTopologyWithInternalTopic(final String firstTableName,
         return builder.build(config);
     }
 
-    @Test
-    public void shouldNotRequireParameters() {
+    @ParameterizedTest(name = "{displayName} - {arguments}")
```

Comment:
What happens if the default `name` is used?

Reply:
```@ParameterizedTest```
```
TopologyTestDriverTest > [1] ENABLED PASSED

TopologyTestDriverTest > [2] DISABLED PASSED

TopologyTestDriverTest > [1] ENABLED PASSED

TopologyTestDriverTest > [2] DISABLED PASSED
```

```@ParameterizedTest(name = "{displayName} - {arguments}")```
```
TopologyTestDriverTest > shouldCloseProcessor(EosMode) - ENABLED PASSED

TopologyTestDriverTest > shouldCloseProcessor(EosMode) - DISABLED PASSED

TopologyTestDriverTest > shouldFeedStoreFromGlobalKTable(EosMode) - ENABLED PASSED

TopologyTestDriverTest > shouldFeedStoreFromGlobalKTable(EosMode) - DISABLED PASSED
```

Reply:
Seems like we should maybe name the enum values: `EOS_ENABLED` and `EOS_DISABLED`, then the output will be good automatically. What do you think? Btw, I am being a nitpicky here because once we have a good pattern, we can apply it in other places.

Reply:
> EOS_ENABLED and EOS_DISABLED

make sense. will copy that.

Reply:
```
TopologyTestDriverTest > shouldCloseProcessor(EosMode) - EOS_ENABLED PASSED

TopologyTestDriverTest > shouldCloseProcessor(EosMode) - EOS_DISABLED PASSED

TopologyTestDriverTest > shouldFeedStoreFromGlobalKTable(EosMode) - EOS_ENABLED PASSED

TopologyTestDriverTest > shouldFeedStoreFromGlobalKTable(EosMode) - EOS_DISABLED PASSED
```

Reply:
I think you can remove the `name` part of `ParamterizedTest` now.

Reply:
copy that!

Reply:
we can't omit the ```name```.

oracle docs:
`It is permissible to omit the element name and equals sign (=) in a single-element annotation whose element name is value`

Reply:
You definitely can, I've done it in many tests in a local branch.

Reply:
I rewrite it to ```@ParameterizedTest("{displayName} - {arguments}")``` and then I get following error.

```
/home/chia7712/kafka/streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java:1918: error: cannot find symbol
    @ParameterizedTest("{displayName} - {arguments}")
                       ^
  symbol:   method value()
  location: @interface ParameterizedTest

```

Reply:
I'm suggesting `ParameterizedTest`, that's it.

Reply:
The output of ```@ParameterizedTest``` is not readable to me so I prefer ```@ParameterizedTest(name = "{displayName} - {arguments}")```. 
```
TopologyTestDriverTest > [1] EOS_ENABLED PASSED

TopologyTestDriverTest > [2] EOS_DISABLED PASSED

TopologyTestDriverTest > [1] EOS_ENABLED PASSED

TopologyTestDriverTest > [2] EOS_DISABLED PASSED

TopologyTestDriverTest > [1] EOS_ENABLED PASSED
```

Reply:
Interesting, I've been testing this with IntelliJ and it seems to handle it differently.

Reply:
I looked into the reasoning for this choice and it's explained here:

https://github.com/junit-team/junit5/issues/1496#issuecomment-404178095

Seems like we are working around the intended design. It would be better to adjust the Gradle config to output the test display name too.

Reply:
Overriding display name by gradle config may cause trouble when the display name from test case is well-designed. Another way is that we can create custom parameterized annotation by junit 5 extension to offer readable display name by default. However, it seems to me that can be addressed in another PR.

I'm fine to use ```@ParameterizedTest``` as junit reports shows both method name and argument name.

Reply:
It looks like a bug to me, personally. Check the Gradle test output html:

![Screenshot from 2021-01-12 13-00-01](https://user-images.githubusercontent.com/24747/104374109-3abafe80-54d6-11eb-9253-38199aa1ea1c.png)

IMO, the text output should output the test name too.

Reply:
It seems to me there are three ways to deal with the repeated magic string (```name = "{displayName} - {arguments}```) in ```ParameterizedTest```

1. write custom ```TestTemplateInvocationContextProvider``` (junit 5 extension). However, the current implementation (```ParameterizedTestExtension```) is a heavy implementation and so it could bring some works to us.
1. junit 5 has to support ```ParameterizedTest``` at class level: https://github.com/junit-team/junit5/issues/878
1. file a PR to junit 5 to make ```ParameterizedTest``` have "long display name" by default , and so we can write following code ```@ParameterizedTest(name = ParameterizedTest.LONG_DISPLAY_NAME)```

At any rate, ```@ParameterizedTest(name = "{displayName} - {arguments}")``` is workaround to us currently and we can improve it in the future If there is any more simple solution for junit 5. 

@ijuma WDYT?

Reply:
I follow your solution (#9874) to remove the ```ParameterizedTest```. see https://github.com/apache/kafka/pull/9856/commits/7480d2692301424e2d9b72e33c1c9b7c34d3c376